src/boot/efi/devicetree.c

```diff
@@ -71,9 +71,10 @@ EFI_STATUS devicetree_install(struct devicetree_state *state, EFI_FILE *root_dir
         assert(root_dir);
         assert(name);
 
+        /* Note: The return value of this function is intentionally not checked. In cases where the system fw
+         * has no devicetree, we still want to load a given devicetree, so a NULL value from this function is
+         * fine. */
         state->orig = find_configuration_table(MAKE_GUID_PTR(EFI_DTB_TABLE));
```

Comment:
Nitpicking but this comment is justifying why we deleted the (broken) error path rather than what the code is actually doing. How about something more like (which, among other things, hints that we did check how the InstallConfigurationTable() in devicetree_cleanup() actually works):

Capture the original value for the device tree table. NULL is not an error in this case so we don't need to check the return value. NULL simply means the system fw had no devicetree initially (and is the correct value to use to return to the initial state if needed).